<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Tamas Schauer" />

<meta name="date" content="2020-05-13" />

<title>Setup DESeq2</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Setup DESeq2</h1>
<h4 class="author">Tamas Schauer</h4>
<h4 class="date">2020-05-13</h4>



<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>
<div id="load-gene-annotation" class="section level2">
<h2>Load Gene Annotation</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(HelpersforDESeq2)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">data_dir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;HelpersforDESeq2&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(rtracklayer)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co"># Annotation rda file was generated from the gtf annotation</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co"># gtf &lt;- import(&quot;dmel-all-r6.17.gtf&quot;)</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co"># save(gtf, file = &quot;dmel-all-r6.17.rda&quot;)</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw">load</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;dmel-all-r6.17.rda&quot;</span>))</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"></a>
<a class="sourceLine" id="cb1-12" data-line-number="12">gtf[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">8</span>]</a></code></pre></div>
<pre><code>## GRanges object with 5 ranges and 8 metadata columns:
##       seqnames            ranges strand |   source     type     score     phase
##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt; &lt;integer&gt;
##   [1]        X 19961297-19969323      + |  FlyBase     gene      &lt;NA&gt;      &lt;NA&gt;
##   [2]        X 19961689-19968479      + |  FlyBase     mRNA      &lt;NA&gt;      &lt;NA&gt;
##   [3]        X 19961689-19961845      + |  FlyBase     5UTR      &lt;NA&gt;      &lt;NA&gt;
##   [4]        X 19961689-19961845      + |  FlyBase     exon      &lt;NA&gt;      &lt;NA&gt;
##   [5]        X 19963955-19964071      + |  FlyBase     exon      &lt;NA&gt;      &lt;NA&gt;
##           gene_id gene_symbol transcript_id transcript_symbol
##       &lt;character&gt; &lt;character&gt;   &lt;character&gt;       &lt;character&gt;
##   [1] FBgn0031081        Nep3          &lt;NA&gt;              &lt;NA&gt;
##   [2] FBgn0031081        Nep3   FBtr0070000           Nep3-RA
##   [3] FBgn0031081        Nep3   FBtr0070000           Nep3-RA
##   [4] FBgn0031081        Nep3   FBtr0070000           Nep3-RA
##   [5] FBgn0031081        Nep3   FBtr0070000           Nep3-RA
##   -------
##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="make-count-table" class="section level2">
<h2>Make Count Table</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">SampleTable &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="kw">file.path</span>(data_dir, <span class="st">&quot;SampleTable.txt&quot;</span>), </a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                           <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">SampleTable &lt;-<span class="st"> </span>SampleTable[<span class="kw">order</span>(SampleTable<span class="op">$</span>ID),]</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">SampleTable</a></code></pre></div>
<pre><code>##     ID Index Barcode   Condition Assay Batch
## 1  IN1     9  GATCAG       Input Input     1
## 14 IP1    11  GGCTAC HighMLEpATP    IP     1
## 2  L15    18  GTCCGC       Input Input     2
## 15 L20    23  GAGTGG HighMLEpATP    IP     2
## 3  L33    13  AGTCAA       Input Input     3
## 18 L34    14  AGTTCC HighMLEmATP    IP     3
## 16 L35    15  ATGTCA HighMLEpATP    IP     3
## 4  L43     1  ATCACG       Input Input     4
## 19 L45     3  TTAGGC HighMLEmATP    IP     4
## 7  L46     5  ACAGTC  LowMLEpATP    IP     4
## 5  L53    13  AGTCAA       Input Input     5
## 8  L54    14  AGTTCC  LowMLEpATP    IP     5
## 12 L55    15  ATGTCA  MedMLEpATP    IP     5
## 17 L56    16  CCGTCC HighMLEpATP    IP     5
## 20 L57    18  GTCCGC HighMLEmATP    IP     5
## 6  L66    19  GTGAAA       Input Input     6
## 10 L67    20  GTGGCC  LowMLEmATP    IP     6
## 11 L68    21  GTTTCG  LowMLEmATP    IP     6
## 9  L69    22  CGTACG  LowMLEpATP    IP     6
## 13 L70    23  GAGTGG  MedMLEpATP    IP     6</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">count_files &lt;-<span class="st"> </span><span class="kw">file.path</span>(data_dir, <span class="kw">list.files</span>(<span class="dt">path =</span> data_dir, <span class="dt">pattern =</span> <span class="st">&quot;ReadsPerGene&quot;</span>))</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">count_names &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.*</span><span class="ch">\\</span><span class="st">/|_[G,A,T,C].*&quot;</span>,<span class="st">&quot;&quot;</span>, count_files)</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="cf">if</span>(<span class="kw">identical</span>(count_names, SampleTable<span class="op">$</span>ID)){</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">        </a>
<a class="sourceLine" id="cb5-6" data-line-number="6">        read.counts &lt;-<span class="st">  </span><span class="kw">makeCountTable</span>(<span class="dt">count_names =</span> count_names, </a>
<a class="sourceLine" id="cb5-7" data-line-number="7">                               <span class="dt">count_files =</span> count_files,</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">                               <span class="dt">stranded =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">}</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="kw">save</span>(read.counts, <span class="dt">file =</span> <span class="kw">paste0</span>(data_dir, <span class="st">&quot;read.counts.rda&quot;</span>))</a>
<a class="sourceLine" id="cb5-12" data-line-number="12"></a>
<a class="sourceLine" id="cb5-13" data-line-number="13">read.counts[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</a></code></pre></div>
<pre><code>##                     IN1      IP1      L15      L20      L33      L34
## N_unmapped     34833971 22851608 25214641 23621948 22811644 16729571
## N_multimapping        0        0        0        0        0        0
## N_noFeature      510171  2491215   367878  2213120   202855   464729
## N_ambiguous      168304   167487    99347   182656   115056   126042
## FBgn0031081           0        1        0        0        0        1
## FBgn0052826           0        0        0        0        0        0
## FBgn0031085           1        0        0        0        0        0
## FBgn0062565          32       49       23       68       15       18
## FBgn0031088           0        0        0        0        0        0
## FBgn0041626           0        3        0        2        0        0</code></pre>
</div>
<div id="setup-deseq2" class="section level2">
<h2>Setup DESeq2</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">library</span>(DESeq2)</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">read.counts.genes &lt;-<span class="st"> </span>read.counts[<span class="op">-</span><span class="dv">1</span><span class="op">:-</span><span class="dv">4</span>,]</a>
<a class="sourceLine" id="cb8-2" data-line-number="2"></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="cf">if</span>(<span class="kw">identical</span>(<span class="kw">colnames</span>(read.counts), SampleTable<span class="op">$</span>ID)){</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">        </a>
<a class="sourceLine" id="cb8-5" data-line-number="5">        dds &lt;-<span class="st"> </span><span class="kw">setupDDS</span>(<span class="dt">CountTableName =</span> <span class="st">&quot;read.counts.genes&quot;</span>, </a>
<a class="sourceLine" id="cb8-6" data-line-number="6">                        <span class="dt">SampleTableName =</span> <span class="st">&quot;SampleTable&quot;</span>,</a>
<a class="sourceLine" id="cb8-7" data-line-number="7">                        <span class="dt">SampleIdName =</span> <span class="st">&quot;ID&quot;</span>, </a>
<a class="sourceLine" id="cb8-8" data-line-number="8">                        <span class="dt">ConditionName =</span> <span class="st">&quot;Condition&quot;</span>, </a>
<a class="sourceLine" id="cb8-9" data-line-number="9">                        <span class="dt">BatchName =</span> <span class="st">&quot;Batch&quot;</span>,</a>
<a class="sourceLine" id="cb8-10" data-line-number="10">                        <span class="dt">n_samples_for_filtering =</span> <span class="kw">ncol</span>(read.counts)<span class="op">*</span><span class="fl">0.75</span>,</a>
<a class="sourceLine" id="cb8-11" data-line-number="11">                        <span class="dt">min_number_of_reads =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb8-12" data-line-number="12">        }</a></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">dds</a></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 8637 20 
## metadata(1): version
## assays(4): counts mu H cooks
## rownames(8637): FBgn0062565 FBgn0031086 ... FBgn0003310 FBgn0031275
## rowData names(58): baseMean baseVar ... deviance maxCooks
## colnames(20): IN1 IP1 ... L69 L70
## colData names(3): Sample Batch sizeFactor</code></pre>
</div>
<div id="get-results" class="section level2">
<h2>Get Results</h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">contrast_list &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;HighMLEpATP&quot;</span>, <span class="st">&quot;Input&quot;</span>),</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">                      <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;LowMLEpATP&quot;</span>,  <span class="st">&quot;Input&quot;</span>),</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">                      <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;HighMLEpATP&quot;</span>, <span class="st">&quot;HighMLEmATP&quot;</span>),</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">                      <span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;LowMLEpATP&quot;</span>, <span class="st">&quot;LowMLEmATP&quot;</span>))</a>
<a class="sourceLine" id="cb17-5" data-line-number="5"></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="cf">for</span>(contrast_name <span class="cf">in</span> contrast_list){</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">        </a>
<a class="sourceLine" id="cb17-8" data-line-number="8">        res &lt;-<span class="st"> </span><span class="kw">getResults</span>(<span class="dt">dds =</span> dds,</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">                          <span class="dt">contrast =</span> contrast_name,</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">                          <span class="dt">lfc_cutoff =</span> <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb17-11" data-line-number="11">                          <span class="dt">shrink =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb17-12" data-line-number="12">                          <span class="dt">annotation =</span> <span class="st">&quot;gtf&quot;</span>,</a>
<a class="sourceLine" id="cb17-13" data-line-number="13">                          <span class="dt">anno_symbol =</span> <span class="st">&quot;gene_symbol&quot;</span>,</a>
<a class="sourceLine" id="cb17-14" data-line-number="14">                          <span class="dt">anno_id =</span> <span class="st">&quot;gene_id&quot;</span>)</a>
<a class="sourceLine" id="cb17-15" data-line-number="15">        </a>
<a class="sourceLine" id="cb17-16" data-line-number="16">        </a>
<a class="sourceLine" id="cb17-17" data-line-number="17">        res_file_name &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;res.&quot;</span>,contrast_name[<span class="dv">2</span>],<span class="st">&quot;-&quot;</span>,contrast_name[<span class="dv">3</span>],<span class="st">&quot;.txt&quot;</span>)</a>
<a class="sourceLine" id="cb17-18" data-line-number="18">        </a>
<a class="sourceLine" id="cb17-19" data-line-number="19">        <span class="kw">write.table</span>(res, <span class="dt">file =</span> <span class="kw">paste0</span>(data_dir, res_file_name), </a>
<a class="sourceLine" id="cb17-20" data-line-number="20">                    <span class="dt">quote =</span> F, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">row.names =</span> T, <span class="dt">col.names =</span> <span class="ot">NA</span>) </a>
<a class="sourceLine" id="cb17-21" data-line-number="21">        </a>
<a class="sourceLine" id="cb17-22" data-line-number="22">}</a>
<a class="sourceLine" id="cb17-23" data-line-number="23"></a>
<a class="sourceLine" id="cb17-24" data-line-number="24"></a>
<a class="sourceLine" id="cb17-25" data-line-number="25"><span class="kw">head</span>(res)</a></code></pre></div>
<pre><code>## log2 fold change (MLE): Sample LowMLEpATP vs LowMLEmATP 
## Wald test p-value: Sample LowMLEpATP vs LowMLEmATP 
## DataFrame with 6 rows and 8 columns
##                     baseMean    log2FoldChange             lfcSE
##                    &lt;numeric&gt;         &lt;numeric&gt;         &lt;numeric&gt;
## FBgn0082957  2474.9854380134  5.35500200795117 0.415969578581414
## FBgn0262871 427.871954829497 -1.47156783886119 0.124240344134104
## FBgn0086661 336.752405379065  2.80397836312988  0.23818771940367
## FBgn0024555  2542.2903537065  -1.3420371461706 0.118975943052019
## FBgn0083046 740.440809016463  4.39764537479637 0.421309579734585
## FBgn0083004 856.641286448432  3.19054079891265 0.315648988785799
##                          stat               pvalue                 padj
##                     &lt;numeric&gt;            &lt;numeric&gt;            &lt;numeric&gt;
## FBgn0082957  12.8735424023396 6.34251516627032e-38 5.47803034910767e-34
## FBgn0262871 -11.8445248129126 2.29716716629638e-32 9.92031640765094e-29
## FBgn0086661  11.7721365742531 5.43308064415868e-32 1.56418391745328e-28
## FBgn0024555 -11.2799034136156 1.64926957041969e-29 3.56118531992871e-26
## FBgn0083046  10.4380379329774  1.6621047892166e-25 2.87111981289275e-22
## FBgn0083004  10.1078758756226 5.09775169197595e-24 7.33821356059938e-21
##                     gene_symbol         chr
##                     &lt;character&gt; &lt;character&gt;
## FBgn0082957 snoRNA:Psi28S-3405d          3R
## FBgn0262871                lute          3R
## FBgn0086661  snoRNA:Psi28S-2566          3L
## FBgn0024555                flfl          3R
## FBgn0083046 snoRNA:Psi18S-1389a          2R
## FBgn0083004 snoRNA:Psi28S-1175b          2R</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] DESeq2_1.24.0               SummarizedExperiment_1.14.1
##  [3] DelayedArray_0.10.0         BiocParallel_1.18.1        
##  [5] matrixStats_0.56.0          Biobase_2.44.0             
##  [7] rtracklayer_1.44.4          GenomicRanges_1.36.1       
##  [9] GenomeInfoDb_1.20.0         IRanges_2.18.3             
## [11] S4Vectors_0.22.1            BiocGenerics_0.30.0        
## [13] HelpersforDESeq2_0.1       
## 
## loaded via a namespace (and not attached):
##  [1] bit64_0.9-7              splines_3.6.1            Formula_1.2-3           
##  [4] assertthat_0.2.1         latticeExtra_0.6-29      blob_1.2.1              
##  [7] GenomeInfoDbData_1.2.1   Rsamtools_2.0.3          yaml_2.2.1              
## [10] RSQLite_2.2.0            pillar_1.4.4             backports_1.1.6         
## [13] lattice_0.20-41          glue_1.4.0               digest_0.6.25           
## [16] RColorBrewer_1.1-2       XVector_0.24.0           checkmate_2.0.0         
## [19] colorspace_1.4-1         htmltools_0.4.0          Matrix_1.2-18           
## [22] XML_3.99-0.3             pkgconfig_2.0.3          genefilter_1.66.0       
## [25] zlibbioc_1.30.0          xtable_1.8-4             purrr_0.3.4             
## [28] scales_1.1.0             jpeg_0.1-8.1             annotate_1.62.0         
## [31] tibble_3.0.1             htmlTable_1.13.3         ggplot2_3.3.0           
## [34] ellipsis_0.3.0           nnet_7.3-14              survival_3.1-12         
## [37] magrittr_1.5             crayon_1.3.4             memoise_1.1.0           
## [40] evaluate_0.14            foreign_0.8-76           tools_3.6.1             
## [43] data.table_1.12.8        lifecycle_0.2.0          stringr_1.4.0           
## [46] locfit_1.5-9.4           munsell_0.5.0            cluster_2.1.0           
## [49] AnnotationDbi_1.46.1     Biostrings_2.52.0        compiler_3.6.1          
## [52] rlang_0.4.6              grid_3.6.1               RCurl_1.98-1.2          
## [55] rstudioapi_0.11          htmlwidgets_1.5.1        bitops_1.0-6            
## [58] base64enc_0.1-3          rmarkdown_2.1            gtable_0.3.0            
## [61] DBI_1.1.0                R6_2.4.1                 GenomicAlignments_1.20.1
## [64] gridExtra_2.3            knitr_1.28               dplyr_0.8.5             
## [67] bit_1.1-15.2             Hmisc_4.4-0              stringi_1.4.6           
## [70] Rcpp_1.0.4.6             geneplotter_1.62.0       vctrs_0.2.4             
## [73] rpart_4.1-15             acepack_1.4.1            png_0.1-7               
## [76] tidyselect_1.0.0         xfun_0.13</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
